src/compile/data/stack.ts

```diff
@@ -1,28 +1,31 @@
-import {DataComponentCompiler} from './base';
-
-import {FacetModel} from './../facet';
-import {LayerModel} from './../layer';
-import {UnitModel} from './../unit';
-
-import {STACKED, SUMMARY} from '../../data';
-import {field, FieldDef} from '../../fielddef';
+import {isArray} from 'vega-util';
+import {field} from '../../fielddef';
 import {hasDiscreteDomain} from '../../scale';
 import {StackOffset} from '../../stack';
 import {contains} from '../../util';
-import {VgData, VgSort, VgTransform} from '../../vega.schema';
+import {VgSort, VgTransform} from '../../vega.schema';
 import {sortParams} from '../common';
+import {Model} from '../model';
+import {UnitModel} from './../unit';
+import {DataFlowNode, DependsOnNode, NewFieldNode} from './dataflow';
 
-export interface StackComponent {
-  /**
-   * Name of the output stacked data source
-   */
-  name: string;
+function getStackByFields(model: Model) {
+  return model.stack.stackBy.reduce((fields, by) => {
+    const channel = by.channel;
+    const fieldDef = by.fieldDef;
 
-  /**
-   * Name of the input source data for stacked data source
-   */
-  source: string;
+    const scale = model.scale(channel);
+    const _field = field(fieldDef, {
+      binSuffix: scale && hasDiscreteDomain(scale.type) ? 'range' : 'start'
+    });
+    if (_field) {
+      fields.push(_field);
+    }
+    return fields;
+  }, [] as string[]);
+}
 
+export interface Stack {
```

Comment:
(Minor) Why don't these become just properties of `StackNode`?
I guess just for convenience?